### DISCUSSION TITLE
Custom Store Notice

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#3777] [Ideas]
In OpenMage we have this option in **Backend > General > Designs > HTML Head**

![store_notice](https://github.com/OpenMage/magento-lts/assets/8360474/095dc00f-4102-4d7a-94ad-1438f7c8d391)

If it is set to yes we see this notice in the header both in Frontend and Backend

![store_message](https://github.com/OpenMage/magento-lts/assets/8360474/84929127-3593-457c-919c-adf7a5a572e1)

I would go ahead and use this notification in production. I would change its name from "Display Demo Store Notice" to "Display Store Notice" option and if Yes is chosen, then a new text field will appear where the text that will be displayed is entered. By default it should be the one for a demo store.

**Comment by addison74** [comment]
It must also be analyzed how the text entered in the box can be translated in any language.

**Reply by sreichel** [reply]
Make that text field based on storeview scope should be fine.

**Comment by Flyingmana** [comment]
If its for a project of you only, then its ok.

If its a plan in general for OpenMage, instead of renaming it, can you add a parallel one?
It has a specific usecase and scope, and from our side we should not shift the scope, just because its similar to what we need.

**Reply by sreichel** [reply]
No need to shift anything ... just allow a custom message.

O/c the default message should be translated in any locale-pack, but why not add the possibility to change it?

**Reply by addison74** [reply]
The change in the OpenMage code offers several advantages to everyone. First of all, the message is editable. Secondly, it is no longer just a demo notification, but can be used for other purposes as well. For example, a maintenance message. I agree there are extensions that offer much more futures, but why couldn't we expand a little what already exists in this case?

**Reply by fballiano** [reply]
I agree with @Flyingmana but the way https://github.com/OpenMage/magento-lts/pull/3778 was done is smart and achieves both the goals

**Comment by sreichel** [comment]
See #3778

**Comment by kiatng** [comment]
I think this is a very useful feature, similar to this [extension](https://github.com/justinstern/magento-notification-bar) that we use.

**Comment by pquerner** [comment]
I think its the themes implementation responsibility to come up with a notification bar, and thus its not OpenMages job to come up with a solution like that.

Also, change this here now means you probably also have to come up with a solution to put it all in a new .phtml file, so the theme can update it if needed.

**Reply by pquerner** [reply]
I want to throw in more things:

Some shops want these in XHR so they can do popups and easier caching.
Some shops want these messages to be more easier maintainable (in terms of editor, and where and when to show such message). Like a CMS page with more features.

So, yes we could add a simply field here and call it a day, or ask why its really needed from the beginning and what are potential use-cases and does this simply field only catch the most basic one?

**Reply by kiatng** [reply]
The idea is to expand the demo store notice, with bit of code changes, it can be used for other types of notices. Sure, with more code and time, more bells and whistles can be added. Just make more PRs.

**Reply by pquerner** [reply]
Oh, theres a dedicated demo store now? Sorry. I have completely missed the point then. My bad.

**Reply by addison74** [reply]
@pquerner - Please review the conversations because it is not about a demo store. I would not like to digress and expand the posts with other topics.

**Reply by pquerner** [reply]
Yea, I was reading it wrong.
Then my first comments still stands.
But I won't stay in the way of course, its just my 2cts.
